lib/puppet/type/user.rb

```diff
@@ -693,7 +693,7 @@ def delimiter
     end
 
     def generate
-      if !self[:purge_ssh_keys].empty? && self[:purge_ssh_keys] != :false
```

Comment:
I think that with the new code `self[:purge_ssh_keys]` is always an array:
https://github.com/puppetlabs/puppet/blob/c4e72a38fad7d768a38a6f80d9a4c435a5ee9b8b/lib/puppet/type/user.rb#L746-L761